File: frontend/lib/api/brand-knowledge.ts
'use client'

import { useQuery } from '@tanstack/react-query'
import { apiClient } from './client'
import type { ApiResponse } from './types'

export type NamedItem = { id: string; name: string; description: string }
export type BrandScope = { workspaceId: string; brandId: string; productId: string }
export type KnowledgePage = {
  id: string; title: string; brand_id: string; product_id: string | null; parent_id: string | null
  kind: 'folder' | 'source' | 'page'; status: 'draft' | 'published' | 'archived'
  revision: number; content?: string; original_name: string | null; updated_at: string
  source_refs: Citation[]
  reused?: boolean
}
export type Citation = {
  page_id: string; title: string; revision: number; excerpt: string; number?: number
  product_id: string | null; kind: string
  product_name?: string | null
}
export type KnowledgeAnswer = { answer: string; citations: Citation[] }
export type ProjectClassification = { id: string; name: string; brand_id: string | null; product_id: string | null }
export const brandRoot = (workspaceId: string) => `/workspaces/${encodeURIComponent(workspaceId)}/brands`
export const knowledgeRoot = (scope: BrandScope) => `${brandRoot(scope.workspaceId)}/${encodeURIComponent(scope.brandId)}`
export const knowledgeParams = (scope: BrandScope) => scope.productId ? { product_id: scope.productId } : {}
export const knowledgeGet = <T,>(path: string, params = {}) =>
  apiClient.get<ApiResponse<T>>(path, { params }).then(result => result.data.data)
export const knowledgePost = <T,>(path: string, body: unknown) =>
  apiClient.post<ApiResponse<T>>(path, body).then(result => result.data.data)

export function useBrands(workspaceId: string) {
  return useQuery({ queryKey: ['brands', workspaceId], enabled: !!workspaceId,
    queryFn: () => knowledgeGet<NamedItem[]>(brandRoot(workspaceId)) })
}
export function useBrandProducts(scope: BrandScope) {
  return useQuery({ queryKey: ['brand-products', scope.workspaceId, scope.brandId], enabled: !!scope.workspaceId && !!scope.brandId,
    queryFn: () => knowledgeGet<NamedItem[]>(`${knowledgeRoot(scope)}/products`) })
}

File: backend/schemas/brand_knowledge.py
from typing import Literal

from pydantic import BaseModel, ConfigDict, Field


class NamedItem(BaseModel):
    model_config = ConfigDict(str_strip_whitespace=True)
    name: str = Field(min_length=1, max_length=120)
    description: str = Field(default="", max_length=4000)


class ProjectClassification(BaseModel):
    product_id: str | None = None


class KnowledgeLibraryCreate(NamedItem):
    pass


class ProjectKnowledgeBindingUpdate(BaseModel):
    product_id: str | None = None


class PageCreate(BaseModel):
    model_config = ConfigDict(str_strip_whitespace=True)
    title: str = Field(min_length=1, max_length=255)
    content: str = Field(default="", max_length=100_000)
    kind: Literal["folder", "page"] = "page"
    parent_id: str | None = None
    product_id: str | None = None


class PageUpdate(BaseModel):
    model_config = ConfigDict(str_strip_whitespace=True)
    title: str = Field(min_length=1, max_length=255)
    content: str = Field(max_length=100_000)
    status: Literal["draft", "published", "archived"]
    revision: int = Field(ge=1)


class KnowledgeQuestion(BaseModel):
    model_config = ConfigDict(str_strip_whitespace=True)
    question: str = Field(min_length=1, max_length=1000)
    product_id: str | None = None
